baselines/flanders/flanders/server.py

```diff
@@ -0,0 +1,360 @@
+
+            # Re-arrange results in the same order as clients' cids impose
+            ordered_results[int(fitres.metrics["cid"])] = (proxy, fitres)
+
+        # Initialize aggregated_parameters if it is the first round
+        if self.aggregated_parameters == []:
+            for key, val in clients_state.items():
+                if val is False:
+                    self.aggregated_parameters = parameters_to_ndarrays(
+                        ordered_results[int(key)][1].parameters
+                    )
+                    break
+
+        # Apply attack function
+        # the server simulates an attacker that controls a fraction of the clients
+        if self.attack_fn is not None and server_round > self.warmup_rounds:
+            log(INFO, "Applying attack function")
+            results, others = self.attack_fn(
+                ordered_results,
+                clients_state,
+                omniscent=self.omniscent,
+                magnitude=self.magnitude,
+                w_re=self.aggregated_parameters,
+                malicious_selected=self.malicious_selected,
+                threshold=self.threshold,
+                d=len(self.aggregated_parameters),
+                old_lambda=self.old_lambda,
+                dataset_name=self.dataset_name,
+                to_keep=self.to_keep,
+                malicious_num=self.num_malicious,
+            )
+            self.old_lambda = others.get("lambda", 0.0)
+
+            # Update saved parameters time series after the attack
+            for _, fitres in results:
+                if fitres.metrics["malicious"]:
+                    if self.sampling > 0:
+                        params = flatten_params(
+                            parameters_to_ndarrays(fitres.parameters)
+                        )[self.params_indexes]
+                    else:
+                        params = flatten_params(
+                            parameters_to_ndarrays(fitres.parameters)
+                        )
+                    log(
+                        INFO,
+                        "Saving parameters of client %s with shape %s",
+                        fitres.metrics["cid"],
+                        params.shape,
+                    )
+                    save_params(
+                        params,
+                        fitres.metrics["cid"],
+                        params_dir=self.history_dir,
+                        remove_last=True,
+                    )
+        else:
+            results = ordered_results
+            others = {}
+
+        # Sort clients states
+        clients_state = {k: clients_state[k] for k in sorted(clients_state)}
+        log(INFO, "Clients state: %s", clients_state)
+
+        # Aggregate training results
+        log(INFO, "fit_round - Aggregating training results")
+        aggregated_result = self.strategy.aggregate_fit(server_round, results, failures)
+
+        if isinstance(self.strategy, Flanders):
```

Comment:
This is needed because `aggregate_fit()` of `FLANDERS` returns extra stuff that other strategies don't. 
I don't know if you need also the return variables to be the same, but I think this is the best solution. The alternative is to remove the extra return values `good_clients_idx` and `malicious_clients_idx` from `FLANDERS` and delegate part of the params saving mechanism to the strategy itself; however I think that splitting this task into two different classes is confusing.

Reply:
Removed in the new version.